src/config/config.service.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from "@nestjs/common";
+import * as dotenv from "dotenv";
+import * as fs from "fs";
+import * as Joi from "@hapi/joi";
+
+
+export type EnvConfig = Record<string, string>;
+
+
+@Injectable()
+export class ConfigService {
+	private readonly envConfig: EnvConfig;
+
+	constructor( filePath: string ) {
+		const config = dotenv.parse( fs.readFileSync( filePath ) );
+		this.envConfig = this.validate( config );
+	}
+
+	private validate( config: EnvConfig ) {
+
+		// Describes Configuration Schema
+		const envVarsSchema: Joi.ObjectSchema = Joi.object( {
+			APP_NAME: Joi.string().default( "Wiredcraft Test API" ),
+			NODE_ENV: Joi.string().valid( "test", "development", "staging", "production" ).default( "development" ),
+			APP_PORT: Joi.number(),
+			MONGO_URI: Joi.string().uri().required(),
+			MONGO_TESTING_URI: Joi.string().uri().required(),
+			JWT_KEY: Joi.string().required()
+		} );
+
+		const { error, value: validatedEnvConfig } = envVarsSchema.validate( config );
+		if ( error ) {
+			throw new Error( `Config Validation Error: ${ error.message }` );
+		}
+		return validatedEnvConfig;
+	}
+
+	public get appName(): string {
+		return this.envConfig.APP_NAME;
+	}
+
+	public get nodeEnv(): string {
+		return process.env.NODE_ENV || this.envConfig.NODE_ENV;
+	}
+
+	public get appPort(): number {
+		return Number( this.envConfig.APP_PORT );
```

Comment:
Yeah, if I add the `convert` flag to the validate call then the types get cast to their intended data type otherwise it stays as a string, then I can skip the type casting. Good catch